--- trunk/gcc/tree-ssa-sccvn.c	2012/09/25 06:58:39	191693
+++ trunk/gcc/tree-ssa-sccvn.c	2012/09/25 07:51:51	191694
@@ -1555,7 +1555,8 @@
 
   /* 3) Assignment from a constant.  We can use folds native encode/interpret
      routines to extract the assigned bits.  */
-  else if (CHAR_BIT == 8 && BITS_PER_UNIT == 8
+  else if (vn_walk_kind == VN_WALKREWRITE
+	   && CHAR_BIT == 8 && BITS_PER_UNIT == 8
 	   && ref->size == maxsize
 	   && maxsize % BITS_PER_UNIT == 0
 	   && offset % BITS_PER_UNIT == 0
